### PR TITLE
Optimize database calls on `storagePoolVolumesGet`

### PR DESCRIPTION
**Pull request by hamistao**
Optimize database calls on `storagePoolVolumesGet` as discussed @tomponline @markylaing @roosterfish

### CODE REVIEW

**Review by roosterfish — CHANGES_REQUESTED** [review]
@hamistao some tests are failing on those changes.

**Comment by hamistao** [comment]
@roosterfish sorry about that, now I believe everything is in order.

File: lxd/db/migration_test.go [hunk 1]
```diff
@@ -147,9 +147,11 @@ func TestImportPreClusteringData(t *testing.T) {
 		volumeType := cluster.StoragePoolVolumeTypeImage
 		filters := []db.StorageVolumeFilter{{
 			Type: &volumeType,
+		}, {
+			PoolID: &id,
```
Comment (tomponline) [inline lxd/db/migration_test.go:151]: This doesn't look quite right? Previously it was filtering by pool id AND volumeType, and now it looks like its filtering by volumeType OR pool id? Am I right?

**Comment by tomponline** [comment]
Any update on this @hamistao ?

File: lxd/db/migration_test.go (hunk 1 above)
Comment (hamistao) [reply lxd/db/migration_test.go:150]: Sorry, I missed the notification for this. Yes, you are correct. I am pushing a new version.

**Review by tomponline — APPROVED** [review]

**Pull request merged**